### ISSUE TITLE
UC 32: As a developer, I want to get the logs of the component I'm working on

### ISSUE THREAD

**Issue by jorgemoralespou** [redhat-developer/odo#177]
TBD

**Comment by kadel** [comment]
it would be nice to have it for MVP at least as `oc` command alias.

**Comment by cdrage** [comment]
@kadel @jorgemoralespou 

I'll work on this.

Do we want *both* BuildConfig and DeploymentConfig logs (I'm assuming so).

**Comment by kadel** [comment]
@cdrage This needs to be broken into smaller well-defined tasks before we start working on it.
Feel free to do it ;-)

We can start with DeploymentConfigs logs. We are already showing BuildConfig logs when a build is started.

**Comment by cdrage** [comment]
@kadel Okay, I'll begin digging through code in order to get the logs from via a library (specifically DeploymentConfig!)

**Comment by kadel** [comment]
There are a few stuff that I think is more important than this. We should fix several painful bugs first.

**Comment by cdrage** [comment]
@kadel Going over how to implement this.

First off, we will simply alias to `oc` for this, until we convert it to it's API counterpart.

What do you think of this?

```sh
odo log --help

Show logs of your component.

Usage:
  odo log [component_name] [flags]

Examples:
  # Get DeploymentConfig and BuildConfig logs for component nodejs (default)
  odo log nodejs

  # Retrieve all logs
  odo log nodejs --all

  # Retrieve only pods and buildconfig
  odo log pods,bc nodejs

Flags:
      --all   Retrieve all associated logs
  -h, --help            help for log


Global Flags:
  -v, --verbose   Verbose output

```

Implementing this would be straight-forward, for example: `odo log pods,bc nodejs` would translate to: `oc get pods,bc nodejs`.

**Comment by kadel** [comment]
odo users shouldn't be exposed to concepts like pods, or buildconfigs.

It should be just `odo log [component name]` if no component name is provided that it would assume current component.

What does `--all` do? 
Will this stream logs or just get all the logs of the component from when it was starting to the time the command is run?

**Comment by cdrage** [comment]
@kadel `--all` would get all the logs deployed by that component.

Streaming could be an option too :+1:

**Comment by cdrage** [comment]
@kadel `odo log` will get buildconfig + deploymentconfig, `--all` would also retrieve Pod logs. Or should we not include Pod logs since it's using the built-in image builder?

**Comment by kadel** [comment]
technically only pods has logs ;-) 

But I don't think we need to show build logs, as they are always displayed when user triggers build via `odo push` or `odo watch`

**Comment by kadel** [comment]
commmand `odo log [component_name]` was added in #429 

Right now it just dumps all component logs to stdout. 

I've created issue https://github.com/redhat-developer/odo/issues/465 to add streaming.

I'm closing this as the basic functionality is done.
